### DISCUSSION TITLE
Build each cs file as it's own bin file on the iso.

### DISCUSSION THREAD

**Discussion by Ollie770** [CosmosOS/Cosmos#2725] [Q&A] [answered]
Hi I'm wondering if it is possible to build each class within my project as its own .bin file on the iso?

So instead of the output being for example "myprojectkernel.bin" it would be "Kernel.bin, App1.bin, CommandLine.bin"
Also if it is possible would all of the output files be linked as if they were a normal build?

The reason behind this is because I test on real hardware and it can be time consuming to rebuild and then burn the iso to my usb.
For example if I was working just in the App1.cs file with no modifications anywhere else I could just build the App1.bin and then replace the one on my usb. Without having to recompile the entire project each time.

Any help would be great Thanks :)

**Comment by zarlo** [comment, marked as answer]
this cant be done with cosmos it would need a large rework of il2cpu

but method level or library level caching is something that could/should be done
